internal/db/collection.go

```diff
@@ -327,7 +327,7 @@ func (c *collection) getAllDocIDsChan(
 
 			canRead, err := c.checkAccessOfDocWithACP(
 				ctx,
-				acp.ReadPermission,
+				acp.PermissionsThatImplyRead,
```

Comment:
note: if this is the reason `CheckDocAccess` checks if ANY permission is given, I strongly disagree with it.
In this particular case you know exactly that you need "read" permission and it's the one you need to pass. I think it's the job of a particular ACP implementation to handle permission inheritance in the best manner. So SourceHub ACP upon handling `CheckDocAccess("read")` should check both "read" and "write" without making other worry about it.

Reply:
- [x] done